src/Symfony/Component/Console/Helper/QuestionHelper.php

```diff
@@ -509,16 +510,22 @@ private function getShell()
         return self::$shell;
     }
 
-    private function isTty(): bool
+    private function isTty(OutputInterface $output): bool
     {
-        $inputStream = !$this->inputStream && \defined('STDIN') ? STDIN : $this->inputStream;
+        if (!$output instanceof StreamOutput) {
+            return true;
```

Comment:
Considering before old mechanism worked differently, this doesn't look right to me. Before, there was no guard condition like this (can be seen in https://github.com/symfony/symfony/pull/33897/files#diff-38eebdda7d6ebfcb7405b91167f05c9cL948), just condition that already is at https://github.com/symfony/symfony/pull/36696/files#diff-312c2e9ca43f41955ae78e8adf50fe49R56

Reply:
implementation updated, please have another look